core/src/gl/mesh.cpp

```diff
@@ -37,17 +37,21 @@ MeshBase::~MeshBase() {
     // Deleting a index/array buffer being used ends up setting up the current vertex/index buffer to 0
     // after the driver finishes using it, force the render state to be 0 for vertex/index buffer
 
-    if (m_glVertexBuffer) {
-        if (RenderState::vertexBuffer.compare(m_glVertexBuffer)) {
-            RenderState::vertexBuffer.init(0, false);
+    if (RenderState::isValidGeneration(m_generation)) {
```

Comment:
Making more work for @blair1618 to rebase/redo!
Refer: https://github.com/tangrams/tangram-es/commit/ba0dcf9bb4f94a383d61aff036cf98107a3308cf